.cirrus.yml

```diff
@@ -94,41 +95,46 @@ win_vm_definition: &WINDOWS_VM_DEFINITION
 # ------------------FILTERS---------------------
 # ----------------------------------------------
 
-only_sonarsource_qa: &ONLY_IF_SONARSOURCE_QA
+only_sonarsource_qa_filter: &ONLY_IF_SONARSOURCE_QA
   only_if: $CIRRUS_USER_COLLABORATOR == 'true' && $CIRRUS_TAG == "" &&
     ($CIRRUS_PR != "" || $CIRRUS_BRANCH == $CIRRUS_DEFAULT_BRANCH || $CIRRUS_BRANCH =~ "branch-.*" || $CIRRUS_BRANCH =~ "dogfood-on-.*")
 
-only_main_branches: &ONLY_MAIN_BRANCHES
+only_main_branches_filter: &ONLY_MAIN_BRANCHES
   only_if: $CIRRUS_USER_COLLABORATOR == 'true' && $CIRRUS_TAG == "" &&
     ($CIRRUS_BRANCH == $CIRRUS_DEFAULT_BRANCH || $CIRRUS_BRANCH =~ "branch-.*")
 
-qa_task_filter_template: &QA_TASK_FILTER
+qa_task_filter: &QA_TASK_FILTER
   depends_on:
     - build
   <<: *ONLY_IF_SONARSOURCE_QA
 
-setup_gradle_cache_template: &SETUP_GRADLE_CACHE
+# ----------------------------------------------
+# ------------------CACHING---------------------
+# ----------------------------------------------
+
+setup_gradle_cache: &SETUP_GRADLE_CACHE
+  create_gradle_directory_script:
+    - mkdir -p ${GRADLE_USER_HOME}
   gradle_cache:
     folder: .gradle/caches
-    fingerprint_script: git rev-parse HEAD
-  create_gradle_directory_script:
-    - mkdir -p "${CIRRUS_WORKING_DIR}/.gradle"
+    fingerprint_script: echo $(date '+%Y-%m-%d')
+    reupload_on_changes: "true"
 
-cleanup_gradle_cache_script_template: &CLEANUP_GRADLE_CACHE_SCRIPT
+cleanup_gradle_cache: &CLEANUP_GRADLE_CACHE
   cleanup_gradle_script:
     - /usr/bin/find "${CIRRUS_WORKING_DIR}/.gradle/caches" -type d -name "8.*" -prune -maxdepth 1 -exec rm -rf {} \;
     - rm -rf "${CIRRUS_WORKING_DIR}/.gradle/caches/journal-1/"
 
-store_profile_report_template: &STORE_PROFILE_REPORT
-  on_success:
-    profile_report_artifacts:
-      path: "build/reports/profile/profile-*.html"
+# ----------------------------------------------
+# -----------------TEMPLATES--------------------
+# ----------------------------------------------
 
 gradle_its_template: &GRADLE_ITS_TEMPLATE
   <<: *SETUP_GRADLE_CACHE
   env:
     # to be able to reuse cached results of previous tasks
     GO_CROSS_COMPILE: '1'
+    KEEP_ORCHESTRATOR_RUNNING: 'true'
```

Comment:
How is the orchestrator stopped then? Is it by `OrchestratorExtension#afterAll` or is it killed with the Cirrus container?

Reply:
It's killed when the container is killed. I had to put such a condition to allow the local execution without the need to kill the process manually. Not perfect but simple.